File: EndGate/EndGate.Core/Collision/CollisionData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EndGate.Core.Assets;

namespace EndGate.Core.Collision
{
    public class CollisionData
    {
        public CollisionData(Vector2d at, Collidable with)
        {
            At = at;
            With = with;
        }

        public Vector2d At { get; private set; }
        public Collidable With { get; private set; }
    }
}

File: EndGate/EndGate.Core.JS/Collision/CollisionData.ts
/// <reference path="../Assets/Vectors/Vector2d.ts" />
/// <reference path="Collidable.ts" />

module EndGate.Collision {

    /**
    * Defines a data object that is used to describe a collision event.
    */
    export class CollisionData {
        /**
        * Who collided with you.
        */
        public With: Collidable;

        /**
        * Creates a new instance of the CollisionData object.
        * @param w Initial value of the With component of CollisionData.
        */
        constructor(w: Collidable) {
            this.With = w;
        }
    }

}